doc/installation.rst

```diff
@@ -1,10 +1,175 @@
-------------
+============
+
+.. todo:: Add ways to use reflex without nix / reflex-plarform
+
+Overview
+--------
+
+The essential components required for developing ``reflex`` based application are
+
+#. GHC or GHCJS
+
+   If you are building a web application with ``reflex-dom`` then you need ``ghcjs`` to create JavaScript output.
+   With ``ghc`` you can use the ``reflex-dom`` to create a ``webkit`` based desktop or mobile app.
+
+#. Reflex library
+
+   The current supported ``reflex`` and ``reflex-dom`` packages (version 0.5 and 0.4 respectively) are available only through Github, as they are not yet published on Hackage.
+
+   To obtain the current ``reflex`` packages, along with all its dependecies, ``reflex-platform`` is the recommended method.
+
+``reflex-platform`` - https://github.com/reflex-frp/reflex-platform
+
+  Is a collection of ``nix`` scripts to provide ``ghc``, ``ghcjs`` and a curated set of packages for use with ``reflex-dom``.
+
+  This includes a specially modified ``text`` package which internally uses the JS string.
+  The performance of this ``text`` package is significantly better on the browser.
+
+You might also need
+
+* Haddock Documentation with local hoogle
+
+* GHCi / ghcid with jsaddle-warp for faster dev cycles
+
+* IDE/Editor support
+
+.. note::
+  GHCJS uses a lot of memory during compilation. 16GB of memory is recommended, with 8GB being pretty close to bare minimum.
+
+.. _reflex_project_skeleton:
+
+``reflex-project-skeleton``
+---------------------------
+
+See - https://github.com/ElvishJerricco/reflex-project-skeleton
+
+This is a bare repository which uses ``reflex-platform`` to provide a nice development enviroment, with both the power of ``nix`` (for binary cache of dependencies) and ``cabal new-*`` commands (for incremental builds).
+
+For a project with both a ``backend`` and ``frontend`` components, this is the recommended setup.
+
+See README and ``reflex-project-skeleton/reflex-platform/project/default.nix`` for usage details.
+
+This also supports cross-compiling the ``frontend`` part to android and iOS platforms!
+
+The following contains information of creating a project-skeleton from scratch and also more details about its working.
+
+https://github.com/reflex-frp/reflex-platform/blob/develop/docs/project-development.md
+
+Minimal dev-env using ``reflex-platform``
+-----------------------------------------
+
+Please refer to `reflex-platform' README <https://github.com/reflex-frp/reflex-platform/blob/develop/README.md#setup>`_
+
+The ``try-reflex`` script can create a development environment with ``ghc`` and ``ghcjs``. You can use this to have a quick starting setup to compile code-snippets and smaller projects.
+
+When using this for the first time, setup can take considerable time to download all the dependencies from the binary cache.
+
+
+But for a non-trivial project it is recommended to use ``cabal``.
+
+
+Using cabal with ``reflex-platform``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
Somewhere in this section we should note that cabal is not to be used for dependency management, as that is Nix's job. Having a bunch of stuff in your `~/.cabal` directory has caused issues in the past. We should also investigate just what those issues are...

Reply:
Added a note about this..